### PR TITLE
Refactor voltage control equations

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactor/Simplify

**What is the new behavior (if this is a feature change)?**
 - BUS_TARGET_V_WITH_SLOPE is removed and replaced by BUS_TARGET_V to be consistent with other voltage control.
 - In AC equation system, BUS_TARGET_V is first and always created so that we don't need to create it in each voltage control and create missing at the end.
 - In AC equation system, code is shared for the 3 voltage control update (code was duplicated and also buggy for transformer).

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java [hunk 1]
```diff
@@ -49,23 +49,22 @@ private void createBusEquation(LfBus bus,
             p.setActive(false);
         }
 
-        createGeneratorControlEquations(bus, equationSystem);
+        // maybe to fix later, but there is so part of OLF (like sensitivity) that needs a voltage target equation
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:52]: This is not equivalent as what we do on main branch. I understand that here you create an inactive equation for all buses. You prefer like this? No performance issue?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:52]: In main we add an if condition on `(!equationSystem.hasEquation(bus.getNum(), AcEquationType.BUS_TARGET_V))`.

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java [hunk 2]
```diff
@@ -91,25 +90,18 @@ private void createGeneratorControlEquations(LfBus bus,
...
             // equation is: V + slope * qSVC = targetV
             // which is modeled here with: V + slope * (sum_branch qBranch) = TargetV - slope * qLoads + slope * qGenerators
-            equationSystem.createEquation(bus, AcEquationType.BUS_TARGET_V_WITH_SLOPE)
-                    .addTerm(vTerm)
+            equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_V).orElseThrow()
                     .addTerms(createReactiveTerms(bus, equationSystem.getVariableSet())
                             .stream()
                             .map(term -> term.multiply(slope))
                             .collect(Collectors.toList()));
-        } else {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:110]: Where are these lines now? You only create the slope equation now.

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:52]: Look at the code, this is already what is done. Creating all V equation at beginning or creating necessary equations and adding missing one at is end is exactly the same.

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:110]: Already created

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java (hunk 1 above)
Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemCreator.java:52]: yes, your are right! All bus_v equations are activated at the end.

**Review by annetill — APPROVED** [review]

**Pull request merged**